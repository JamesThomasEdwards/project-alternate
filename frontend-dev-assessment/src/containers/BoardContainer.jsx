// React;
import React from 'react';
import { Component } from 'react';
// Files;
import Board from '../components/Board.jsx';
import boardContentJSONString from '../../boardContentJSONString';

export default class BoardContainer extends Component {
    state = {
        headerContent: [],
        showHidePanelContent: [[true, null], [false, null]],
        showHideHeaderContent: true,
        boardContent: null
    };
    // toggles panel content, showing text/ background lightend on one;
    // and darkend background with no text on the other;
    togglePanelContent = (id) => {
        let boardContent = this.state.boardContent;
        if (id === 1) {
            this.setState(() => ({
                showHidePanelContent: [[false, null], [true, boardContent.contentB]]
            }));
        } else {
            this.setState(() => ({
                showHidePanelContent: [[true, boardContent.contentA], [false, null]]
            }));
        };
    };
    // toggles header content;
    // press on the '-' to hide content and change '-' to '+'; 
    // '+' to show content and change back to '-'
    toggleHeaderContent = () => {
        let boardContent = this.state.boardContent;
        if (this.state.showHideHeaderContent) {
            this.setState(() => ({
                headerContent: [],
                showHideHeaderContent: false
            }));
        } else {
            this.setState(() => ({
                headerContent: boardContent.headerData,
                showHideHeaderContent: true
            }));
        };
    };
    // takes in the JSON string, parse it, and updates the state;
    componentDidMount() {
        let boardContent = JSON.parse(boardContentJSONString);
        this.setState(() => ({
            headerContent: boardContent.headerData,
            showHidePanelContent: [[true, boardContent.contentA], [false, null]],
            boardContent
        }));
    }
    render() {
        return (
            <div>
                <Board
                    headerData={this.state.headerContent}
                    toggleHeaderContent={this.toggleHeaderContent}
                    showHideHeaderContent={this.state.showHideHeaderContent}
                    showHidePanelContent={this.state.showHidePanelContent}
                    togglePanelContent={this.togglePanelContent} />
            </div>
        );
    };
};
