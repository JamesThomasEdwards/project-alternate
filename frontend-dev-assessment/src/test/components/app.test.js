import React from 'react'
import App from '../../app.jsx';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import BoardContainer from '../../containers/BoardContainer.jsx';
import { configure } from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('should render App component correctly', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<App />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

