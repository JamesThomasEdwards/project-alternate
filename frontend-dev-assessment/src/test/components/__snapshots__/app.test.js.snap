// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component correctly 1`] = `
<div>
  <div
    className="boards-container"
  >
    <BoardContainer
      id={0}
    />
  </div>
  <div
    className="boards-container"
  >
    <BoardContainer
      id={1}
    />
  </div>
  <div
    className="boards-container"
  >
    <BoardContainer
      id={2}
    />
  </div>
</div>
`;
